### PR TITLE
GH-41791: [CI][Conda] Update azure.linux.yml task, replace CondaEnvironment@1 with Bash@3

### PR DESCRIPTION
**Pull request by amoeba**
### What changes are included in this PR?

- Updated azure.linux.yml task to remove what looks like an unneeded step
- Updated macros.ninja to replace use of CondaEnvironment@1 with an equivalent Bash@3 task

### Are these changes tested?

Testing in CI.

### Are there any user-facing changes?

No
* GitHub Issue: #41791

### CODE REVIEW

**Comment by amoeba** [comment]
@github-actions crossbow submit -g conda

**Comment by amoeba** [comment]
Hrm. Is there actually a good way to test this? The bit I updated is guarded by an `arrow.is_default_branch()` but I also don't want to mess anything up on Anaconda.

File: dev/tasks/macros.jinja [hunk 1]
```diff
@@ -220,15 +220,13 @@ env:
...
-      packageSpecs: 'anaconda-client'
-      installOptions: '-c conda-forge'
-      updateConda: no
-  - script: |
-      conda install -y anaconda-client
-      anaconda -t $(CROSSBOW_ANACONDA_TOKEN) upload --force {{ pattern }}
-    displayName: Upload packages to Anaconda
+      targetType: inline
+      script: |
+        conda create -y -n azure_upload_anaconda -c conda-forge anaconda-client
```
Comment (amoeba) [inline dev/tasks/macros.jinja:227]: Could probably just install into the base environment to save time.

**Comment by kou** [comment]
What is the blocker of this?
It seems that we can start building with Docker but it's failed. (I think that it's out-of-scope of this.)

**Comment by amoeba** [comment]
Before merging, I wanted to make sure my change actually works. I think I can just create a test pipeline on Azure to make sure my change does what I think it will do. I'll report back once that runs.

**Comment by amoeba** [comment]
I tested a similar set of commands in a new Azure pipeline and found a couple of issues. https://github.com/apache/arrow/pull/41883/commits/b307165aedc16e3b2f2c775aa6f89a1b51694806 includes those fixes so I think this will run correct now. @kou do you want to take a look?

**Review by kou — APPROVED** [review]
+1

File: dev/tasks/macros.jinja [hunk 2]
```diff
@@ -220,15 +220,10 @@ env:
...
-      installOptions: '-c conda-forge'
-      updateConda: no
-  - script: |
-      conda install -y anaconda-client
-      anaconda -t $(CROSSBOW_ANACONDA_TOKEN) upload --force {{ pattern }}
-    displayName: Upload packages to Anaconda
+  - bash: |
+        conda create -y -n azure_upload_anaconda -c conda-forge anaconda-client
+        source activate azure_upload_anaconda
+        anaconda -t $(CROSSBOW_ANACONDA_TOKEN) upload --force {{ pattern }}
```
Comment (kou) [inline dev/tasks/macros.jinja:226]: Could you unindent them?

```suggestion
      conda create -y -n azure_upload_anaconda -c conda-forge anaconda-client
      source activate azure_upload_anaconda
      anaconda -t $(CROSSBOW_ANACONDA_TOKEN) upload --force {{ pattern }}
```

**Comment by amoeba** [comment]
Merged, thank you as always @kou.

**Pull request merged**